NightscoutUploadKit/NightscoutUploader.swift

```diff
@@ -8,22 +8,43 @@
 
 import Crypto
 
-public enum UploadError: Error {
+public enum UploadError: LocalizedError {
     case httpError(status: Int, body: String)
     case missingTimezone
     case invalidResponse(reason: String)
     case unauthorized
     case missingConfiguration
     case invalidParameters
```

Comment:
This helps to describe the errors in Caregiver

Reply:
I am curious if any feedback on this error paradigm in general -- using LocalizedErrors and `throws`. The remote commands 2.0 will have the potential for deeply nested errors in Swift concurrent APIs, as there are many things that can cause a failure in distant code. My current approach has been to define Localized errors like this. Then most methods use `throws`, which gets propogated to the bottom of the callstack where I present whatever error was thrown -- without any knowledge of its type. 

I mention this as I see a lot of Loop APIs that use Result Types and specialized error types for error handling -- that are not using Swift Concurrency. I think returning the Result type with specialized errors can work as well but the code seems cleaner/simpler when you can just place a `try` before failable calls, rather than thinking about what to do with each result (i.e. with guards, Result unwrapping, early returns, etc.)

When you see some of the remote 2.0 code, the question may make more sense and we can revisit it then if not clear.